app/use_cases/recalculate_scores.rb

```diff
@@ -24,7 +24,9 @@ def execute
 
   def recalculate_response_ids(response_ids)
     ResponseContent.where(:id.in => response_ids).pluck(:content, :id).each do |content, rcid|
-      next if content.nil? # Can theoretically happen
+      # Note that we don't check for csrf_failed here (on purpose)
+      # It doesn't hurt to calculate scores for csrf_failed responses.
```

Comment:
are you sure? it would hurt the calculation in terms of invalid data right?

Reply:
The score calculation has it built-in that if certain scores are missing, it won't calculate the total score, if you set `require_all` to `true`. If you don't set `require_all`, it will still calculate the scores as long as there is at least one of the scores used present.

What I meant was is that it doesn't affect the distributions since the scores are calculated just for the response itself, and affect only the response itself.